### PR TITLE
contrib/go-redis/redis.v8: support go-redis.v8

### PR DESCRIPTION
**Pull request by sfroment**
This is simply a port of contrib/go-redis/redis, but for go-redis.v8 with the new Hook mechanism.

### CODE REVIEW

**Review by knusbaum — CHANGES_REQUESTED** [review]
Thanks for this. I only have a few comments.

**Comment by sfroment** [comment]
Sorry for the delay updated the review, do not hesitate if there is anything else that need to me done :)

**Review by knusbaum — DISMISSED** [review]
LGTM. Thank you @sfroment

File: contrib/go-redis/redis.v8/redis.go [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+
+	"github.com/go-redis/redis/v8"
+)
+
+type datadogHook struct {
+	*params
+}
```
Comment (johejo) [inline contrib/go-redis/redis.v8/redis.go:28]: I think we should check the implementation of the Hook interface at compile time.
```
var _ redis.Hook = (*datadogHook)(nil)
```

File: contrib/go-redis/redis.v8/redis.go [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+	args := cmd.Args()
+	if len(args) == 0 {
+		return ""
+	}
+	if str, ok := args[0].(string); ok {
+		return str
+	}
+	return ""
+}
```
Comment (johejo) [inline contrib/go-redis/redis.v8/redis.go:162]: Cmder implements fmt.Stringer at v8. https://pkg.go.dev/github.com/go-redis/redis/v8#Cmder
We don't need this hack.

File: contrib/go-redis/redis.v8/redis.go [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+
+	"github.com/go-redis/redis/v8"
+)
+
+type datadogHook struct {
+	*params
+}
+
+var _ redis.Hook = (*datadogHook)(nil)
```
Comment (knusbaum) [inline contrib/go-redis/redis.v8/redis.go:30]: Generally we put these checks in the test files.

**Review by knusbaum — CHANGES_REQUESTED** [review]

**Review by knusbaum — DISMISSED** [review]
This looks great. Thanks.

@sfroment Since `v1.27.0` was released, the PR needs to be rebased on `v1` in order to get the tests to pass. 
I am happy to rebase this for you if you don't have time.

**Review by knusbaum — APPROVED** [review]

**Pull request merged**